llpc/tool/amdllpc.cpp

```diff
@@ -452,54 +452,72 @@ static Result Init(
             }
         }
 
-        // Initialize the path for shader cache
-        constexpr uint32_t MaxFilePathLen = 512;
-        char               shaderCacheFileDirOption[MaxFilePathLen];
+        const char* pName = "-shader-cache-file-dir";
+        size_t nameLen = strlen(pName);
+        bool found = false;
+        const char* pArg = nullptr;
+        for (int32_t i = 1; i < argc; ++i)
+        {
+            pArg = argv[i];
+            if ((strncmp(pArg, pName, nameLen) == 0) &&
```

Comment:
you can declare `name` as `llvm::StringLiteral` and use make `pArg` be `llvm::StringRef args` and check if if starts with what you expect by calling `arg.starts_with`